File: python/examples/markdown_viewer/blocks/heading.py
"""Heading block for the markdown viewer."""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))

from heading_ui import HeadingUI


HEADING_COLORS = {
    1: "#facc15",
    2: "#93c5fd",
    3: "#a7f3d0",
}


class Heading(HeadingUI):
    def set_heading(self, level, text):
        marker = "#" * max(1, min(3, int(level)))
        color = HEADING_COLORS.get(max(1, min(3, int(level))), "#dbeafe")
        self.heading_text.set_spans([
            (marker, "#64748b"),
            (" ", None),
            (text, color),
        ])

File: cpp/examples/markdown_viewer/blocks/heading.cpp
#include "heading.hpp"

#include <algorithm>
#include <utility>

namespace markdown_viewer_example {

void Heading::setHeading(int level, std::string value) {
    const int clamped = std::max(1, std::min(3, level));
    const std::string color = clamped == 1 ? "#facc15" : (clamped == 2 ? "#93c5fd" : "#a7f3d0");
    heading_text->setSpans({
        ui::LabelSpan{.text = std::string(static_cast<std::size_t>(clamped), '#'), .foreground = "#64748b"},
        ui::LabelSpan{.text = " "},
        ui::LabelSpan{.text = std::move(value), .foreground = color},
    });
}

}  // namespace markdown_viewer_example
